pages/common/options-loader.ts

```diff
@@ -65,7 +73,7 @@ export function useOptionsLoader<Item>({ pageSize = 25, timeout = 1000, randomEr
           reject();
         } else {
           const nextItems = sourceItems.slice(pageNumber * pageSize, (pageNumber + 1) * pageSize);
-          resolve({ items: nextItems, hasNextPage: items.length + nextItems.length < sourceItems.length });
+          resolve({ items: nextItems, hasNextPage: itemsRef.current.length + nextItems.length < sourceItems.length });
```

Comment:
This fixes an issue when hasNextPage is set to false because the check is done against the prev state. As result, there is sometimes a bug in test pages when only 1st page of items is loaded.